File: src/quiz_app/models/question.py
"""Data models for the quiz application."""

from dataclasses import dataclass


@dataclass
class Question:
    """Represents a single exam question loaded from the JSON."""

    id: str
    text: str
    options: dict[str, str]
    correct_answers: list[str]

    @property
    def is_multiple_choice(self) -> bool:
        """Returns True if there is more than one correct answer."""
        return len(self.correct_answers) > 1

File: android-app/app/src/main/java/com/example/quizapp/model/Question.kt
package com.example.quizapp.model

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class Question(
    val id: String,
    val text: String,
    val options: Map<String, String>,
    @SerialName("correct_answers")
    val correctAnswers: List<String>
) {
    fun isMultipleChoice(): Boolean = correctAnswers.size > 1
}
